File: sentimental-credit/credit.py
from cs50 import get_int

credit = str(get_int("Number: "))
first2 = credit[0] + credit[1]
length = len(credit)
multiply_digits = 0
normal_digits = 0

for i in range(length):
    # Accessed credit number be reversed
    number = int(credit[length-1-i])
    # Get every last other digits
    if i % 2 == 1:
        x2 = number*2
        # If there number more than 9 iterate on each digit's and sum them
        if x2 > 9:
            tmp = str(x2)
            x2 = 0
            for i in tmp:
                x2 += int(i)
        # The digits are multiply by 2
        multiply_digits += x2
    else:
        # The odd normal digits
        normal_digits += number
result = multiply_digits+normal_digits
if (result % 10) == 0:
    if first2 in ['34', '37'] and length == 15:
        print("AMEX")
    elif first2[0] == '4' and length in [13, 16]:
        print("VISA")
    elif int(first2) >= 51 and int(first2) <= 55:
        print("MASTERCARD")
    else:
        print("INVALID")
else:
    print("INVALID")

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int product_digits(int n);

int main(void)
{
    string digits = get_string("Number: ");
    int credit_length = strlen(digits);
    char first2[] = {digits[0], digits[1]};
    int first2_numbers = atoi(first2);
    int sum_numbers_multiply_2 = 0;
    int sum_numbers_not_multiply_2 = 0;
    char revers_digits[20];
    for (int i = credit_length - 1; i >= 0; i--)
    {
        revers_digits[(credit_length - 1) - i] = digits[i];
    }

    for (int i = 0; i < credit_length; i++)
    {
        int current_number = revers_digits[i] - '0';
        if (i % 2 == 1)
        {
            int digit = current_number * 2;
            if (digit > 9)
            {
                sum_numbers_multiply_2 += product_digits(digit);
            }
            else
            {
                sum_numbers_multiply_2 += digit;
            }
        }
        else
        {
            sum_numbers_not_multiply_2 += current_number;
        }
    }
    int final_digit = (sum_numbers_not_multiply_2 + sum_numbers_multiply_2) % 10;
    if (final_digit == 0)
    {
        if ((first2_numbers == 34 || first2_numbers == 37) && credit_length == 15)
        {
            printf("AMEX\n");
        }
        else if (first2[0] == '4' && (credit_length == 13 || credit_length == 16))
        {
            printf("VISA\n");
        }
        else if ((first2_numbers >= 51 && first2_numbers <= 55) && credit_length == 16)
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

int product_digits(int n)
{
    char digits[2];
    int sum = 0;
    sprintf(digits, "%i", n);
    int len = strlen(digits);
    for (int i = 0; i < len; i++)
    {
        sum += digits[i] - '0';
    }
    return sum;
}
